yarn-project/cli/src/cmds/l1/update_l1_validators.ts

```diff
@@ -0,0 +1,153 @@
+import { EthCheatCodes } from '@aztec/aztec.js';
+import { ETHEREUM_SLOT_DURATION, type EthAddress } from '@aztec/circuits.js';
+import { createEthereumChain } from '@aztec/ethereum';
+import { type DebugLogger, type LogFn } from '@aztec/foundation/log';
+import { RollupAbi } from '@aztec/l1-artifacts';
+
+import { createPublicClient, createWalletClient, getContract, http } from 'viem';
+import { generatePrivateKey, mnemonicToAccount, privateKeyToAccount } from 'viem/accounts';
+
+export interface RollupCommandArgs {
+  rpcUrl: string;
+  chainId: number;
+  privateKey: string | undefined;
+  mnemonic: string;
+  rollupAddress: EthAddress;
+}
+
+export interface LoggerArgs {
+  log: LogFn;
+  debugLogger: DebugLogger;
+}
+
+export function generateL1Account() {
+  const privateKey = generatePrivateKey();
+  const account = privateKeyToAccount(privateKey);
+  account.address;
+  return {
+    privateKey,
+    address: account.address,
+  };
+}
+
+export async function addL1Validator({
+  rpcUrl,
+  chainId,
+  privateKey,
+  mnemonic,
+  validatorAddress,
+  rollupAddress,
+  log,
+  debugLogger,
+}: RollupCommandArgs & LoggerArgs & { validatorAddress: EthAddress }) {
+  const dualLog = makeDualLog(log, debugLogger);
+  const { walletClient, publicClient } = getL1Clients(rpcUrl, chainId, privateKey, mnemonic);
+  const rollup = getContract({
+    address: rollupAddress.toString(),
+    abi: RollupAbi,
+    client: walletClient,
+  });
+
+  dualLog(`Adding validator ${validatorAddress.toString()} to rollup ${rollupAddress.toString()}`);
+  const txHash = await rollup.write.addValidator([validatorAddress.toString()]);
+  dualLog(`Transaction hash: ${txHash}`);
+  await publicClient.waitForTransactionReceipt({ hash: txHash });
+  dualLog(`Funding validator on L1`);
+  const cheatCodes = new EthCheatCodes(rpcUrl, debugLogger);
+  await cheatCodes.setBalance(validatorAddress, 100000000000000000000n);
```

Comment:
Notation wise, might be easier if using `100n*10n**18n` since easier to get the number of zeros.